packages/core/src/VDocument.ts

```diff
@@ -98,18 +102,19 @@ export class VDocument {
      * orphaned children into the parent of the first removed node.
      */
     deleteSelection(): void {
+        const range = this.selection.range;
         withMarkers(() => {
-            const nodes = this.range.selectedNodes;
+            const nodes = range.selectedNodes;
             if (!nodes.length) return;
-            this.range.collapse(this.range.start); // Reset the direction of the range.
-            let reference = this.range.end;
+            range.collapse();
```

Comment:
Is this still needed if we don't handle the direction specifically? It seems that we were collapsing only to reset the direction.